components/org.wso2.carbon.identity.idp.metadata.saml2/src/main/java/org/wso2/carbon/identity/idp/metadata/saml2/util/SAML2SSOFederatedAuthenticatorConfigBuilder.java

```diff
@@ -62,260 +80,261 @@ private static FederatedAuthenticatorConfig parse(EntityDescriptor entityDescrip
+                }
+                properties[0] = property;
+
+                property = new Property();
+                property.setName(IdentityApplicationConstants.Authenticator.SAML2SSO.SP_ENTITY_ID);
+                property.setValue(""); // Not available in the metadata specification.
+                properties[1] = property;
+
+                property = new Property();
+                property.setName(IdentityApplicationConstants.Authenticator.SAML2SSO.SSO_URL);
+                List<SingleSignOnService> singleSignOnServices = idpssoDescriptor.getSingleSignOnServices();
+                if (CollectionUtils.isNotEmpty(singleSignOnServices)) {
+                    boolean found = false;
+                    for (SingleSignOnService singleSignOnService : singleSignOnServices) {
+                        if (singleSignOnService != null) {
+                            if (singleSignOnService.getLocation() != null) {
+                                property.setValue(singleSignOnService.getLocation());
+                                found = true;
+                                break;
                             }
                         }
-                        if (!found) {
-                            property.setValue("");
-                            throw new IdentityApplicationManagementException("No SSO URL, invalid file content");
-                        }
-                    } else {
+                    }
+                    if (!found) {
                         property.setValue("");
                         throw new IdentityApplicationManagementException("No SSO URL, invalid file content");
                     }
-                    properties[2] = property;
-
-                    property = new Property();
-                    property.setName(IdentityApplicationConstants.Authenticator.SAML2SSO.IS_AUTHN_REQ_SIGNED);
-                    if (idpssoDescriptor.getWantAuthnRequestsSigned() != null && idpssoDescriptor.getWantAuthnRequestsSigned() == true) {
-                        property.setValue("true");
-                    } else {
-                        property.setValue("false");
-                    }
-                    properties[3] = property;
-
-                    property = new Property();
-                    property.setName(IdentityApplicationConstants.Authenticator.SAML2SSO.IS_LOGOUT_ENABLED);
-                    List<SingleLogoutService> singleLogoutServices = idpssoDescriptor.getSingleLogoutServices();
-                    if (CollectionUtils.isNotEmpty(singleLogoutServices)) {
-                        property.setValue("true");
-                    } else {
-                        property.setValue("false");
+                } else {
+                    property.setValue("");
+                    throw new IdentityApplicationManagementException("No SSO URL, invalid file content");
+                }
```

Comment:
You don't need this `else` as well right?

Reply:
Will open separate PR for the fix and checkstyle issues then it is easier for review.